### ISSUE TITLE
PNPM still makes symbolic links with node-linker=hoisted 

### ISSUE THREAD

**Issue by EddyVinck** [pnpm/pnpm#6754]
### pnpm version:
8.2.0 and 8.6.5

### Code to reproduce the issue:

```
pnpm config list
# node-linker=hoisted
# registry=https://registry.npmjs.org/
# strict-peer-dependencies=false
# user-agent=pnpm/8.2.0 npm/? node/v18.1.0 darwin arm64
```

```
pnpm install
```
### Expected behavior:

No symbolic links, just copies of the workspaces in node_modules

According to [the documentation](https://pnpm.io/npmrc#node-linker):

> hoisted - a flat node_modules without symlinks is created

### Actual behavior:

Symbolic links are produced (notice the arrows to the right of `@acme/db` and `esbuild`.

<img width="253" alt="image" src="https://github.com/pnpm/pnpm/assets/23434753/7517c72c-9013-407b-9946-2024ec733d08">

### Additional information:

 - `node -v` prints: 19.7.0
 - Windows, macOS, or Linux?: MacOS

**Comment by zkochan** [comment]
What is the target directory of the symlink?

**Comment by zkochan** [comment]
Even if you use `node-linker=hoisted`, local workspace dependencies are symlinked. This works the same way with Yarn and npm too, in monorepos. If you don't wan't symlinks even in this case, you can use the injected feature: https://pnpm.io/package_json#dependenciesmetainjected

**Comment by EddyVinck** [comment]
@zkochan Thanks! I hadn't seen that feature yet. In the meantime I found a workaround, but it's good to know the option is there.

Would it be possible to inject workspace projects with a flag for `pnpm install`? If not, I think it could be a great feature to have.

**Comment by ennioVisco** [comment]
> @zkochan Thanks! I hadn't seen that feature yet. In the meantime I found a workaround, but it's good to know the option is there.
> 
> Would it be possible to inject workspace projects with a flag for `pnpm install`? If not, I think it could be a great feature to have.

I agree, I also found it very confusing. I get the retro-compatibility, but it doesn't seem to make a lot of sense. Perhaps it did in an era where monorepo were still being explored?

**Comment by zkochan** [comment]
> Would it be possible to inject workspace projects with a flag for pnpm install? If not, I think it could be a great feature to have.

I guess so

**Comment by BoscoDomingo** [comment]
How did this end? I'm setting up a monorepo deployment and pnpm insists on symlinking my local library, even when using `pnpm fetch` and `pnpm deploy`, breaking dependencies when I run individual apps

**Comment by CosmoWorker** [comment]
@EddyVinck could you please say the workaround you did to get over this issue, in my turborepo (using pnpm) prisma client & other are being sym linked, what do I do?

**Comment by zkochan** [comment]
Closing: current pnpm covers this.

Workspace projects are linked whatever the node linker; `nodeLinker: hoisted` decides how the dependencies from the registry are laid out, not how the projects of the workspace are wired together.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
